isteven-multi-select.js

```diff
@@ -1,1111 +1,1377 @@
-                                if ( typeof value[ attrs.groupProperty ] === 'undefined' ) {                                
-                                    value[ $scope.tickProperty ] = true;
-                                }
-                            }
-                        });                            
-                        $scope.refreshOutputModel();                                    
-                        $scope.refreshButton();                                                  
-                        $scope.onSelectAll();                                                
-                        break;
-                    case 'NONE':
-                        angular.forEach( $scope.filteredModel, function( value, key ) {
-                            if ( typeof value !== 'undefined' && value[ attrs.disableProperty ] !== true ) {                        
-                                if ( typeof value[ attrs.groupProperty ] === 'undefined' ) {                                
-                                    value[ $scope.tickProperty ] = false;
-                                }
-                            }
-                        });               
-                        $scope.refreshOutputModel();                                    
-                        $scope.refreshButton();                                                                          
-                        $scope.onSelectNone();                        
-                        break;
-                    case 'RESET':            
-                        angular.forEach( $scope.filteredModel, function( value, key ) {                            
-                            if ( typeof value[ attrs.groupProperty ] === 'undefined' && typeof value !== 'undefined' && value[ attrs.disableProperty ] !== true ) {                        
-                                var temp = value[ $scope.indexProperty ];                                
-                                value[ $scope.tickProperty ] = $scope.backUp[ temp ][ $scope.tickProperty ];
-                            }
-                        });               
-                        $scope.refreshOutputModel();                                    
-                        $scope.refreshButton();                                                                          
-                        $scope.onReset();                        
-                        break;
-                    case 'CLEAR':
-                        $scope.tabIndex = $scope.tabIndex + 1;
-                        $scope.onClear();    
-                        break;
-                    case 'FILTER':                        
-                        $scope.tabIndex = helperItems.length - 1;
-                        break;
-                    default:                        
-                }                                
```

Comment:
vs-repeat div is used to encapsulate the ng-repeat. This is for scroll and rendering performance